### PR TITLE
feat(forge): install with no args fetches submodules

### PR DESCRIPTION
**Pull request by regynald**
Addresses #558. Fetches submodules without grabbing latest commits like `forge update` does.

I originally implemented this with a new command `forge fetch` to fetch a single submodule instead of all of them. Not sure which implementation is preferred.

### CODE REVIEW

File: cli/src/cmd/install.rs [hunk 1]
```diff
@@ -42,6 +42,12 @@ pub(crate) fn install(
     dependencies: Vec<Dependency>,
     opts: DependencyInstallOpts,
 ) -> eyre::Result<()> {
+    if dependencies.is_empty() {
+        let mut cmd = Command::new("git");
+        cmd.args(&["submodule", "update", "--init", "--recursive"]);
+        cmd.spawn()?.wait()?;
+    }
```
Comment (mattsse) [inline cli/src/cmd/install.rs:49]: great

**Review by mattsse — APPROVED** [review]

File: cli/README.md [hunk 2]
```diff
@@ -27,6 +27,7 @@ SUBCOMMANDS:
     clean              removes the build artifacts and cache directories
     completions        generate shell completions script
     create             deploy a compiled contract
+    fetch              fetches upstream libs without grabbing new commits
```
Comment (mattsse) [inline cli/README.md:30]: no longer needed?

File: cli/src/cmd/install.rs (hunk 1 above)
Comment (gakonst) [reply cli/src/cmd/install.rs:49]: Yep that's my understanding, right @brockelmore ?

Comment (brockelmore) [reply cli/src/cmd/install.rs:49]: yes that's the thought (if there are no new deps being added)

File: cli/README.md (hunk 2 above)
Comment (regynald) [reply cli/README.md]: good catch. ty ty

**Pull request merged**